src/types.jl

```diff
@@ -269,12 +286,17 @@ end
 # `step` instead of `state`.  The step would contain the current state
 # of the solution: `(t,y)` at minimum, but it could also be
 # `(t,y,dy,dt)`.  Thoughts?
+#
+#m3: No, that doesn't work if we want to allow zero-allocation
+#    algorithms.  Unless you make `step` part of `state` but then it
+#    becomes pointless.
+
```

Comment:
What about `accept!(solver,state,step)` instead of `accept!(solver,state)`?  Any stepper will have `state` contains `step` anyway.